zerver/tests/test_signup.py

```diff
@@ -2549,6 +2550,76 @@ def test_access_signup_page_in_root_domain_without_realm(self) -> None:
         result = self.client_get('/register', subdomain="", follow=True)
         self.assert_in_success_response(["Find your Zulip accounts"], result)
 
+    def test_signup_with_accepted_invitation(self) -> None:
+        self.user_profile = self.example_user('iago')
+        stream_name = "Denmark"
+        streams = [get_stream(stream_name, self.user_profile.realm)]
+
+        do_invite_users(self.user_profile, ["foo@zulip.com"], streams, False)
+        prereg_user = PreregistrationUser.objects.get(email="foo@zulip.com")
+
+        key = generate_key()
+        Confirmation.objects.create(content_object=prereg_user,
+                                    date_sent=timezone_now(),
+                                    confirmation_key=key,
+                                    type=Confirmation.INVITATION)
+
+        self.assertEqual(prereg_user.status, 0)
+        url = confirmation_url(key, self.user_profile.realm.host, Confirmation.INVITATION)
+        response = self.client_get(url)
+
+        prereg_user.refresh_from_db()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(prereg_user.status, 1)
+
+        info = {
+            'full_name': prereg_user.full_name,
+            'password': "password",
+            'key': key,
+            'terms': True,
+        }
+        self.client_post('/accounts/register/', info)
+        prereg_user.refresh_from_db()
+        self.assertEqual(prereg_user.status, 1)
+
+    def test_signup_with_accepted_invitation_as_staff_with_support_param(self) -> None:
+        self.user_profile = self.example_user('iago')
+        self.user_profile.is_staff = True
+        self.login(self.example_email('iago'))
+
+        stream_name = "Denmark"
+        streams = [get_stream(stream_name, self.user_profile.realm)]
+
+        do_invite_users(self.user_profile, [self.nonreg_email('alice')], streams, False)
+        prereg_user = PreregistrationUser.objects.create(email=self.nonreg_email('alice'),
+                                                         referred_by=self.user_profile,
+                                                         realm=self.user_profile.realm)
+
+        key = generate_key()
+        Confirmation.objects.create(content_object=prereg_user,
+                                    date_sent=timezone_now(),
+                                    confirmation_key=key,
+                                    type=Confirmation.INVITATION)
+
+        self.assertEqual(prereg_user.status, 0)
+        url = confirmation_url(key, self.user_profile.realm.host, Confirmation.INVITATION)
```

Comment:
Instead of doing all this can you use the `invite` helper function in `InviteUserTest`?